src/lib/y2network/config_writer/sysconfig.rb

```diff
@@ -36,21 +36,22 @@ def write(config)
 
         write_ip_forwarding(config.routing)
 
-        # list of devices used by routes
-        devices = config.interfaces + [nil]
-
-        devices.each do |dev|
+        # Write ifroute files
+        config.interfaces.each do |dev|
           routes = find_routes_for(dev, config.routing.routes)
           file = routes_file_for(dev)
 
-          # Remove ifroutes-* if empty, but not /etc/sysconfig/network/route
-          if dev && routes.empty?
-            file.remove
-          else
-            file.routes = routes
-            file.save
-          end
+          # Remove ifroutes-* if empty
+          file.remove if  routes.empty?
+
+          file.routes = routes
+          file.save
         end
+
+        # update /etc/sysconfig/network/routes file
+        file = routes_file_for(nil)
```

Comment:
using `nil` here is not much expressive. What about a symbol or something like that?